### PR TITLE
 modified product module

### PR DESCRIPTION
**Pull request by Avetiq30**

### CODE REVIEW

File: src/order/order.model/order.model.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ProductModel } from 'src/products/product.model/product.model';
+import { ProductModel } from 'src/product/model/product-model';
```
Comment (VahagShahinyan) [inline src/order/order.model/order.model.ts:1]: change import path to ```./```

File: src/product/dto/product-filter.dto.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  @IsOptional()
+  @Transform(({ value }) => parseInt(value))
+  @IsNumber()
+  maxPrice: number;
+
+  @IsOptional()
+  @IsString()
+  @IsNotEmpty()
+  category: string;
```
Comment (VahagShahinyan) [inline src/product/dto/product-filter.dto.ts:18]: Change ```category``` to ```categoryName```

File: src/product/model/product-model.ts
Comment (VahagShahinyan) [inline src/product/model/product-model.ts:1]: Rename ```product-model.ts``` to ```product.model.ts```

File: src/product/prdouct-constants.ts
Comment (VahagShahinyan) [inline src/product/prdouct-constants.ts:1]: Rename  ```prdouct-constants.ts``` to ``` prdouct.constant.ts```

File: src/product/products.module.ts
Comment (VahagShahinyan) [inline src/product/products.module.ts:1]: Rename ```products.module.ts``` to ```product.module.ts```

Comment (VahagShahinyan) [reply src/product/products.module.ts:1]: Renam  ```export class ProductsModule {}``` to ```export class ProductModule {}```

File: src/product/prdouct-constants.ts [hunk 3]
```diff
@@ -1,2 +1,3 @@
 export const PRODUCT_NOT_FOUND = 'Product not found';
 export const NOT_FOUND_CATEGORY = 'Category not found';
+export const NOT_FOUN_PRODUCT_BY_ID = 'Could not found product by id';
```
Comment (VahagShahinyan) [inline src/product/prdouct-constants.ts:3]: Rename  ```NOT_FOUN_PRODUCT_BY_ID``` to ```NOT_FOUND_PRODUCT_BY_ID```

File: src/order/order.model/order.model.ts (hunk 1 above)
Comment (Avetiq30) [reply src/order/order.model/order.model.ts:1]: ++

File: src/product/dto/product-filter.dto.ts (hunk 2 above)
Comment (Avetiq30) [reply src/product/dto/product-filter.dto.ts:18]: ++

File: src/product/model/product-model.ts
Comment (Avetiq30) [reply src/product/model/product-model.ts:1]: ++

File: src/product/prdouct-constants.ts
Comment (Avetiq30) [reply src/product/prdouct-constants.ts:1]: ++

File: src/product/products.controller.ts
[the remark this answers is not in the document]
Comment (Avetiq30) [reply src/product/products.controller.ts:1]: ++

File: src/product/products.module.ts
Comment (Avetiq30) [reply src/product/products.module.ts:1]: ++

File: src/product/prdouct-constants.ts (hunk 3 above)
Comment (Avetiq30) [reply src/product/prdouct-constants.ts:3]: ++

**Pull request merged**